src/org/traccar/smpp/ClientSmppSessionHandler.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.smpp;
+
+import org.traccar.helper.Log;
+
+import com.cloudhopper.commons.charset.CharsetUtil;
+import com.cloudhopper.smpp.SmppConstants;
+import com.cloudhopper.smpp.impl.DefaultSmppSessionHandler;
+import com.cloudhopper.smpp.pdu.DeliverSm;
+import com.cloudhopper.smpp.pdu.PduRequest;
+import com.cloudhopper.smpp.pdu.PduResponse;
+
+public class ClientSmppSessionHandler extends DefaultSmppSessionHandler {
+
+    private SmppClient smppClient;
+
+    public ClientSmppSessionHandler(SmppClient smppClient) {
+        this.smppClient = smppClient;
+    }
+
+    @Override
+    public void firePduRequestExpired(PduRequest pduRequest) {
+        Log.warning("PDU request expired: " + pduRequest);
+    }
+
+    @Override
+    public PduResponse firePduRequestReceived(PduRequest request) {
+        PduResponse response = null;
+        try {
+            if (request instanceof DeliverSm) {
+                if (request.getOptionalParameters() != null) {
+                    Log.debug("Message Delivered: "
```

Comment:
Now delivery reports look like this:
````
2017-02-22 16:13:50 DEBUG: Message Delivered: 9433835d-cbe4-4838-b13e-34f31dacba70, State: 2
````

Where `State` is https://github.com/fizzed/cloudhopper-smpp/blob/master/src/main/java/com/cloudhopper/smpp/SmppConstants.java#L288